controllers/chat/moveToAnon.js

```diff
@@ -34,6 +34,17 @@ const moveToAnon = async (req, res) => {
   if (!members.includes(req.userId)) members.push(req.userId);
 
   const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+
+  const filter = {id: oldChannelId};
+  const sort = {last_message_at: -1};
+  const oldChannelDetail = await client.queryChannels(filter, sort);
+  //req.userId always anon since the token are using anon token
+  const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
+  const prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
+    (user) => user.user_id !== mySignedId
+  )[0];
+  const detailPrevTargetUser = await UsersFunction.findUserById(User, prevTargetUser.user_id);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is querying the old channel details and then filtering out the current user to get the target user. This could be optimized by directly querying for the target user instead of getting all users and then filtering out the current user.
```diff
-  const filter = {id: oldChannelId};
-  const sort = {last_message_at: -1};
-  const oldChannelDetail = await client.queryChannels(filter, sort);
-  //req.userId always anon since the token are using anon token
-  const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
-  const prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
-    (user) => user.user_id !== mySignedId
-  )[0];
-  const detailPrevTargetUser = await UsersFunction.findUserById(User, prevTargetUser.user_id);

+  const targetUserId = members.find(member => member !== req.userId);
+  const detailPrevTargetUser = await UsersFunction.findUserById(User, targetUserId);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->